node-modules-polyfill/src/index.ts

```diff
@@ -91,6 +87,33 @@ export function NodeModulesPolyfillPlugin(
                     .map(escapeStringRegexp)
                     .join('|'), // TODO builtins could end with slash, keep in mind in regex
             )
+
+            const prefixFilter = new RegExp(`^.*\0polyfill-node\.`)
+            onResolve({ filter: prefixFilter }, (args) => {
+                const path = args.path.replace(prefixFilter, '')
+                const isCommonjs =
+                    args.namespace !== commonjsNamespace &&
+                    args.kind === 'require-call'
```

Comment:
I copied this from the other resolver but didn't really take the time to understand fully what it is doing so may need some consideration.